modules/mobile/app/models/mobile/v0/adapters/appointment_requests.rb

```diff
@@ -0,0 +1,208 @@
+# frozen_string_literal: true
+
+module Mobile
+  module V0
+    module Adapters
+      class AppointmentRequests
+        # accepts an array of appointment requests
+        # returns a list of appointments
+        # only returns SUBMITTED appointment requests
+        # and CANCELLED requests that were created in the past 30 days
+        def parse(requests)
+          va_appointments = []
+          cc_appointments = []
+
+          requests.each do |request|
+            status = status(request)
+            next unless status.in?(%w[CANCELLED SUBMITTED])
+
+            if status == 'CANCELLED'
+              created_at = Date.strptime(request[:created_date], '%m/%d/%Y')
+              next if created_at < 30.days.ago
+            end
+
+            klass = request.key?(:cc_appointment_request) ? CC : VA
+            model = build_appointment_model(request, klass)
+
+            if klass == VA
+              va_appointments << model
+            else
+              cc_appointments << model
+            end
+          end
+
+          [va_appointments, cc_appointments]
+        end
+
+        private
+
+        # rubocop:disable Metrics/MethodLength
+        def build_appointment_model(request, klass)
+          start_date = localized_start_date(request)
+
+          Mobile::V0::Appointment.new(
+            id: request[:appointment_request_id],
+            appointment_type: klass::APPOINTMENT_TYPE,
+            cancel_id: nil,
+            comment: nil,
+            facility_id: klass.facility_id(request),
+            sta6aid: nil,
+            healthcare_provider: klass.provider_name(request),
+            healthcare_service: klass.practice_name(request),
+            location: klass.location(request),
+            minutes_duration: nil,
+            phone_only: phone_only?(request),
+            start_date_local: start_date,
+            start_date_utc: start_date.utc,
+            status: status(request),
+            status_detail: nil,
+            time_zone: time_zone(request),
+            vetext_id: nil,
+            reason: request[:purpose_of_visit],
+            is_covid_vaccine: nil, # unable to find or create test data for this
```

Comment:
None of the fixture data contains covid vaccinations. I tried creating covid vaccination appointment requests in the web app but they never appeared in the list data. I tried adding a "useCache: false" param, but it didn't change anything. I checked again the next day but again nothing changed. We almost surely can know this from "visit type" but I need to figure out how to get better data.

Reply:
judy

Reply:
Interesting. So when you make a covid vaccine request, it bypasses pending and becomes an actual appointment?